DSharpPlus/Net/WebSocket/WebSocketClient.cs

```diff
@@ -291,50 +292,46 @@ public static IWebSocketClient CreateNew(IWebProxy proxy)
         /// </summary>
-        public event AsyncEventHandler Connected
+        public event AsyncEventHandler<IWebSocketClient, SocketEventArgs> Connected
         {
             add => this._connected.Register(value);
             remove => this._connected.Unregister(value);
         }
-        private AsyncEvent _connected;
+        private AsyncEvent<WebSocketClient, SocketEventArgs> _connected;
 
         /// <summary>
         /// Triggered when the client is disconnected.
         /// </summary>
-        public event AsyncEventHandler<SocketCloseEventArgs> Disconnected
+        public event AsyncEventHandler<IWebSocketClient, SocketCloseEventArgs> Disconnected
         {
             add => this._disconnected.Register(value);
             remove => this._disconnected.Unregister(value);
         }
-        private AsyncEvent<SocketCloseEventArgs> _disconnected;
+        private AsyncEvent<WebSocketClient, SocketCloseEventArgs> _disconnected;
 
         /// <summary>
         /// Triggered when the client receives a message from the remote party.
         /// </summary>
-        public event AsyncEventHandler<SocketMessageEventArgs> MessageReceived
+        public event AsyncEventHandler<IWebSocketClient, SocketMessageEventArgs> MessageReceived
         {
             add => this._messageReceived.Register(value);
             remove => this._messageReceived.Unregister(value);
         }
-        private AsyncEvent<SocketMessageEventArgs> _messageReceived;
+        private AsyncEvent<WebSocketClient, SocketMessageEventArgs> _messageReceived;
 
         /// <summary>
         /// Triggered when an error occurs in the client.
         /// </summary>
-        public event AsyncEventHandler<SocketErrorEventArgs> ExceptionThrown
+        public event AsyncEventHandler<IWebSocketClient, SocketErrorEventArgs> ExceptionThrown
         {
             add => this._exceptionThrown.Register(value);
             remove => this._exceptionThrown.Unregister(value);
         }
-        private AsyncEvent<SocketErrorEventArgs> _exceptionThrown;
+        private AsyncEvent<WebSocketClient, SocketErrorEventArgs> _exceptionThrown;
 
-        private void EventErrorHandler(string evname, Exception ex)
-        {
-            if (evname.ToLowerInvariant() == "ws_error")
-                Console.WriteLine($"WSERROR: {ex.GetType()} in {evname}!");
-            else
-                this._exceptionThrown.InvokeAsync(new SocketErrorEventArgs(null) { Exception = ex }).ConfigureAwait(false).GetAwaiter().GetResult();
-        }
+        private void EventErrorHandler<TArgs>(AsyncEvent<WebSocketClient, TArgs> asyncEvent, Exception ex, AsyncEventHandler<WebSocketClient, TArgs> handler, WebSocketClient sender, TArgs eventArgs)
+            where TArgs : AsyncEventArgs
+            => this._exceptionThrown.InvokeAsync(this, new SocketErrorEventArgs() { Exception = ex }).ConfigureAwait(false).GetAwaiter().GetResult();
```

Comment:
Why does error handling need to be handled synchronously compared to the other event invokers?

Reply:
The upstream intent here was logging etc. The previous implementation did the same.